src/libcore/ops.rs

```diff
@@ -1530,6 +1533,75 @@ impl<Idx: fmt::Debug> fmt::Debug for RangeTo<Idx> {
     }
 }
 
+/// An inclusive range which is bounded at both ends.
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
+pub enum RangeInclusive<Idx> {
+    /// Empty range (iteration has finished)
+    Empty {
+        /// The point at which iteration finished
+        at: Idx
+    },
+    /// Non-empty range (iteration will yield value(s))
+    NonEmpty {
```

Comment:
One point of discomfort about this representation: it can easily be a lie if you write down an empty range.